### ISSUE TITLE
Maximum Update Depth Exceeded bug

### ISSUE THREAD

**Issue by TaiWilkin** [open-apparel-registry/open-apparel-registry#1297]
Going to the facilities list and clicking on a facility causes a `Maximum update depth exceeded` error in the developer tools, and the page freezes. It doesn't occur when going to the facility directly. This error is present on Staging but not Production.

**Comment by caseycesari** [comment]
It doesn't happen every time, but I've run into this when visiting a facility page directly.

**Comment by jwalgran** [comment]
I added `console.log` statements inside each of the the `useEffect` handlers in our `useUpdateLeafletMapImperatively` hook. After clicking a facility in the results list I saw the last `useEffect`, responsible for panning the map to the selected location, stuck in a loop. Commenting out https://github.com/open-apparel-registry/open-apparel-registry/blob/b220433dfc476ac147d96ef3e6c26a3bc3ec187a/src/app/src/util/hooks.js#L123-L162 (and adding `/* eslint-disable no-unused-vars */
`) broke the map panning, but the update depth error and unresponsiveness when clicking Claim this facility (#1392) were resolved.

I will now look deeper into why this `useEffect` is not behaving the way we intend it to.
